test/topology_experimental_raft/test_mv_tablets.py

```diff
@@ -7,15 +7,74 @@
 # Tests for interaction of materialized views with *tablets*
 
 from test.pylib.manager_client import ManagerClient
+from test.pylib.util import wait_for_cql_and_get_hosts, read_barrier
+from test.pylib.internal_types import ServerInfo
+from test.topology.conftest import skip_mode
 
 import pytest
 import asyncio
 import logging
 import random
+import time
 
 
 logger = logging.getLogger(__name__)
 
+# This convenience function takes the name of a table or a view, and a token,
+# and returns the list of host_id,shard pairs holding tablets for this token
+# and view.
+# You also need to specify a specific server to use for the requests, to
+# ensure that if you send tablet-migration commands to one server, you also
+# read the replicas information from the same server (it takes time for this
+# information to propagate to all servers).
+async def get_tablet_replicas(manager: ManagerClient, server: ServerInfo, keyspace_name: str, table_or_view_name: str, token: int):
+    host = (await wait_for_cql_and_get_hosts(manager.cql, [server], time.time() + 60))[0]
+    await read_barrier(manager.cql, host)
+
+    # "table_or_view_name" may be either a table or a view, and those are
+    # listed in different system tables so we may need to search both:
+    matches = list(await manager.cql.run_async(f"select id from system_schema.tables where keyspace_name = '{keyspace_name}' and table_name = '{table_or_view_name}'"))
+    if not matches:
+        matches = list(await manager.cql.run_async(f"select id from system_schema.views where keyspace_name = '{keyspace_name}' and view_name = '{table_or_view_name}'"))
```

Comment:
You could extend ManagerClient.get_table_id().

Reply:
That's exactly why I don't like putting convenience functions in "global" places - unless very well thought out, when the next user wants to use them they never do the right thing, and then they start to get 17 different options and features that people needed in 17 different tests. (again, this is my bad experience from dtest speaking..).

In this case, as you guessed I tried to use get_table_id() but it wasn't good enough because it didn't search for views. You're right I could have improved get_table_id() but the question is - why. Why should the get_table_id() functionality exist in the "manager"? Why is it the job of the cluster manager to provide 100 different convenience functions that perform specific CQL requests?

Reply:
By the way, I just realized that this function didn't need to use the concept of "table_id" at all this complicated way to fetch it. If could just look for the given `table_name` and `keyspace_name` (but need to use `ALLOW FILTERING`, this becomes a whole-table scan of the tablets table). This would find both tables and view names.

(Another by the way: If you do use table_id like you do today, you don't need to also filter the keyspace when reading the tablets table - a table_id can't possibly belong to other keyspaces).

Reply:
I decided to stop using table_id at all (so in particular I won't need get_table_id()). I'll send that change as a followup patch which will also improve the Alternator LSI tests (to use 2 nodes instead of 10).

Reply:
> In this case, as you guessed I tried to use get_table_id() but it wasn't good enough because it didn't search for views. You're right I could have improved get_table_id() but the question is - why. Why should the get_table_id() functionality exist in the "manager"? Why is it the job of the cluster manager to provide 100 different convenience functions that perform specific CQL requests?

Why extend get_table_id() - because new logic fits into its semantics. 

Why have get_table_id() - It's good to have common vocabulary for working with scylla, so that tests are easier to write, read and maintain. When I see get_table_id() I immediately know what is intended. When writing a test, I don't have to figure out the implementation on my own (column names?). It's also more robust against changes of the system.  For example, you added another implementation of get_tablet_replicas(). If we change the layout of system.tablets, we now have to update it in two places rather than one.

Why manager - it doesn't have to live in the manager, it could live somewhere else. It's common to all tests which work with scylla.

> (Another by the way: If you do use table_id like you do today, you don't need to also filter the keyspace when reading the tablets table - a table_id can't possibly belong to other keyspaces).

Keyspace name is part of primary key, so it had to be specififed.

Reply:
I changed the test - https://github.com/scylladb/scylladb/pull/16440/commits/a24c16333865937b68bc2409e34561762711d01f - to completely avoid a table_id

Basically you can just look up (with ALLOW FILTERING - this is a full table scan) given table_name and keyspace_name - no need to look up the table_id. Maybe the table_id is useful when there might be thousands of tables in the system but this doesn't happen in our test.

Reply:
I wonder if we want to follow up with the changelog note re this change. If you agree it's the case, please update @tzach